sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -535,17 +528,15 @@ public virtual async Task<Response<DownloadManifestResult>> DownloadManifestAsyn
 
                 rawResponse.Headers.TryGetValue("Docker-Content-Digest", out var digest);
 
-                if (!ValidateDigest(rawResponse.ContentStream, digest))
-                {
-                    throw new RequestFailedException("The requested digest does not match the digest of the received manifest.");
-                }
+                var contentDigest = BlobHelper.ComputeDigest(rawResponse.ContentStream);
+                ValidateDigest(contentDigest, digest);
 
-                using var document = JsonDocument.Parse(rawResponse.ContentStream);
+                using var document = JsonDocument.Parse(rawResponse.Content);
                 var manifest = OciManifest.DeserializeOciManifest(document.RootElement);
 
                 rawResponse.ContentStream.Position = 0;
 
-                return Response.FromValue(new DownloadManifestResult(digest, manifest, rawResponse.ContentStream), rawResponse);
+                return Response.FromValue(new DownloadManifestResult(digest, manifest, rawResponse.Content), rawResponse);
```

Comment:
nit and future: we should probably deserialize content into manifest lazily inside `DownloadManifestResult` when user requests it.

Reply:
I'll add this to the Manifest API work - good suggestion - thank you!

Reply:
Tracked here: https://github.com/Azure/azure-sdk-for-net/issues/23828#issuecomment-1396205254